frontend/pages/queries/details/QueryDetailsPage/QueryDetailsPage.tsx

```diff
@@ -201,11 +201,22 @@ const QueryDetailsPage = ({
   const isApiError = storedQueryError || queryReportError;
   const isClipped =
     (queryReport?.results?.length ?? 0) >= QUERY_REPORT_RESULTS_LIMIT;
-  const disabledLiveQuery = config?.server_settings.live_query_disabled;
+  const isLiveQueryDisabled = config?.server_settings.live_query_disabled;
 
   const renderHeader = () => {
+    // Team admins/maintainers can only edit queries assigned to a team
     const canEditQuery =
-      isGlobalAdmin || isGlobalMaintainer || isTeamMaintainerOrTeamAdmin;
+      isGlobalAdmin ||
+      isGlobalMaintainer ||
+      (isTeamMaintainerOrTeamAdmin && storedQuery?.team_id);
+
+    const canLiveQuery =
+      lastEditedQueryObserverCanRun ||
+      isObserverPlus ||
+      isAnyTeamObserverPlus ||
+      isGlobalAdmin ||
+      isGlobalMaintainer ||
+      isTeamMaintainerOrTeamAdmin;
```

Comment:
same as above - does this check that the permission aligns with the query's team_id?

Reply:
for this one `canLiveQuery`, is almost any user (all listed) except observer, but observer can run live query if it is set to observer can run